graphql-java-servlet/src/main/java/graphql/kickstart/servlet/BatchedQueryResponseWriter.java

```diff
@@ -34,6 +41,16 @@ public void write(HttpServletRequest request, HttpServletResponse response) thro
 
     String responseContent = responseBuilder.toString();
     byte[] contentBytes = responseContent.getBytes(StandardCharsets.UTF_8);
+
+    if (responseCache != null && responseCache.isCacheable(request, invocationInput)) {
```

Comment:
This block or something nearly identical is repeated in every `QueryResponseWriter` implementation. Instead it would be better if you'd apply the decorator pattern. If somebody enabled caching then have the `QueryResponseWriter` factory method create some `CachingResponseWriter` (which should implement `QueryResponseWriter` too obviously) wrapping the other implementations. That way you have one class with this logic and you don't pollute the other classes with caching logic. Ideally they're not aware of that at all.

Reply:
Done.